### ISSUE TITLE
Feature Request in Layers: Store best model with save_best_only.

### ISSUE THREAD

**Issue by cpury** [tensorflow/tfjs#296]
#### TensorFlow.js version
0.10.3

#### Browser version
Latest Chrome

#### Describe the problem or feature request
When training on a just a few samples as usually happens in the browser, training and validation loss tend to jump around a bit.

It would be great if we could automatically get the best model after training finishes.

This is possible in Keras with the `ModelCheckpoint(save_best_only=True)` callback and afterwards loading the weights from that saved file.

Alternatively, if I knew how to copy a model in its current state, it would be straightforward to write the logic for this in a `onEpochEnd` callback.

**Comment by bileschi** [comment]
Great suggestion!  FYI, model copying is coming soon.  Please follow https://github.com/tensorflow/tfjs-core/pull/1038 for guidance on that front.

**Comment by cpury** [comment]
@bileschi it seems model copying is now implemented, which only implements this feature indirectly.

Is there a best practice / guide to retain the best model in training? That would also be a useful addition to the docs, I believe.

**Comment by bileschi** [comment]
There is not such a guide or demo at this point.  Agreed that it would make an excellent addition to the tutorials.

**Comment by cpury** [comment]
@bileschi I went ahead and tried it out. Works great! Thanks for pointing me to the other issue.

Here's what I did:

```
let bestEpoch = -1;
let bestValLoss = Number.MAX_SAFE_INTEGER;
const bestModelPath = 'localstorage://best-model'

model.fit(..., {
	...,
	callbacks: {
		onEpochEnd: async function (epoch, logs) {
			if (logs.val_loss < bestValLoss) {
				// Store best model:
				bestEpoch = epoch;
				bestValLoss = logs.val_loss;
				await training.currentModel.save(bestModelPath);
			}
		},
		onTrainEnd: async function () {
			// Load best model:
			model = await tf.loadModel(bestModelPath);
		},
	}
});
```

Note that if you want to keep training afterwards, you need to `.compile()` the model again.

**Comment by cpury** [comment]
Quick question: Does TF.JS `await` the callbacks? Otherwise there might be a race condition here.
